src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c

```diff
@@ -65,122 +149,999 @@ static DamageTable D_80BA82CC = {
+            } else if (!(player->stateFlags2 & 0x80)) {
+                func_800BDC5C(&this->skelAnime, sAnimations, 1);
+                this->actor.flags |= 1;
+                this->unk_3F0 = 2;
+                this->unk_3F2 = 0;
+            }
+            break;
+
+        case 1:
+            if (this->unk_3F2 == 20) {
+                sp32 = player->ageProperties->unk_92 + 0x6805;
+                globalCtx->damagePlayer(globalCtx, -8);
+                func_800B8E58(&player->actor, sp32);
+                func_8013ECE0(this->actor.xzDistToPlayer, 240, 1, 12);
+                this->unk_3F2 = 0;
+                // if (player){};
+            } else {
+                // if (player){};
+                this->unk_3F2++;
+            }
+
+            // if (player){};
+            if (func_801378B8(&this->skelAnime, 0.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_ATTACK);
+            }
+
+            if (!(player->stateFlags2 & 0x80) || (player->unk_B62 != 0)) {
+                if ((player->unk_B62 != 0) && (player->stateFlags2 & 0x80)) {
+                    player->stateFlags2 &= ~0x80;
+                    player->unk_AE8 = 100;
+                }
+                func_800BDC5C(&this->skelAnime, sAnimations, 1);
+                this->actor.flags |= 1;
+                this->unk_3F0 = 2;
+                this->unk_3F2 = 0;
+            }
+            break;
+
+        case 2:
+            if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+                this->unk_3F4 = 40;
+                this->actor.shape.yOffset = 0.0f;
+                func_80BA5AF0(this);
+            } else {
+                Math_SmoothStepToF(&this->actor.shape.yOffset, 0.0f, 1.0f, 400.0f, 0.0f);
+            }
+            break;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Railgibud/func_80BA5E18.s")
+#endif
+
+void func_80BA6054(EnRailgibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 7);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DAMAGE);
+    this->actor.speedXZ = -2.0f;
+    this->actionFunc = func_80BA60B0;
+}
+
+void func_80BA60B0(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->actor.speedXZ < 0.0f) {
+        this->actor.speedXZ += 0.15f;
+    }
+
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    Math_SmoothStepToS(&this->unk_3E2, 0, 1, 0x12C, 0);
+    Math_SmoothStepToS(&this->unk_3E8, 0, 1, 0x12C, 0);
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        func_80BA6158(this);
+    }
+}
+
+void func_80BA6158(EnRailgibud* this) {
+    this->unk_3F2 = 0;
+    func_800BDC5C(&this->skelAnime, sAnimations, 10);
+    this->actionFunc = func_80BA61A0;
+}
+
+void func_80BA61A0(EnRailgibud* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.world.rot.y, BINANG_ROT180(this->actor.yawTowardsPlayer), 5, 0xDAC, 0xC8);
```

Comment:
```suggestion
    Math_SmoothStepToS(&this->actor.world.rot.y, BINANG_ROT180(this->actor.yawTowardsPlayer), 5, 3500, 200);
```

Reply:
Why are these in decimals? In other PRs I had to keep values dealing with rotation in hex. Why decimal this time?

Reply:
As I was told, Different actor devs chose to use fairly roundish numbers but their preference of hex vs dec varied.

I was told, if you can make it a round number in hex or dec, then use that.

previous discussion:
https://github.com/zeldaret/mm/pull/62#discussion_r596560985